### PR TITLE
Initial commit for Windows hardlinking.

### PR DESCRIPTION
**Pull request by tokejepsen**
Have a look at the README for usage details.

Any thoughts big or small are welcome :)

### CODE REVIEW

File: file_link/hardlink.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import platform
+
+
+def create_windows(source, destination):
```
Comment (mottosso) [inline file_link/hardlink.py]: I would prepend an underscore to highlight that these aren't meant to be used externally.

File: file_link/hardlink.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    """Creates hardlink at destination from source in Linux."""
+    raise NotImplementedError("Linux is not support yet.")
+
+
+def create_osx(source, destination):
+    """Creates hardlink at destination from source in OSX."""
+    raise NotImplementedError("OSX is not support yet.")
+
+
+def create(source, destination):
```
Comment (mottosso) [inline file_link/hardlink.py]: I would call this simply `link` and make the arguments `source` and `link_name` so we align with [`os.link`](https://docs.python.org/2/library/os.html#os.link). It's not the prettiest argument names, but better to align with something than nothing.

File: file_link/hardlink.py (hunk 1 above)
Comment (tokejepsen) [reply file_link/hardlink.py]: Good point! Done.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,19 @@
...
+## Goal
+
+To provide a simple module for linking files on disk to different locations.
+
+## Usage
+
+This module is as simple as
+
+```python
+from file_link import hardlink
```
Comment (mottosso) [inline README.md]: How about..

```python
import file_link
file_link.link("/some/file", "/linked/file")
```

Though I'm no fan of the double "link".

File: file_link/hardlink.py (hunk 2 above)
Comment (tokejepsen) [reply file_link/hardlink.py]: Ohh, you don't think the the end user clearity outweighs alignment?

File: README.md (hunk 3 above)
Comment (mottosso) [reply README.md]: How about `filelink` without the underscore, to mimic `tempfile`?

```python
import filelink
filelink.create("..", "..")
```

Comment (tokejepsen) [reply README.md]: Maybe..
```python
import file_link.hard("/some/file", "/hard_linked/file")
import file_link.sym("/some/file", "/sym_linked/file")
```

File: file_link/hardlink.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    system = platform.system()
+
+    if system == "Windows":
+        _create_windows(source, destination)
+
+    if system == "Linux":
+        _create_linux(source, destination)
+
+    if system == "Darwin":
```
Comment (mottosso) [inline file_link/hardlink.py]: Should probably make these `elif` and end with `"Unsupported platform"`.

Comment (tokejepsen) [reply file_link/hardlink.py]: Yup, agree. Done.

File: file_link/hardlink.py (hunk 2 above)
Comment (mottosso) [reply file_link/hardlink.py]: About the arguments, I'm thinking we can either (1) restart comprehension with new, clean names like the ones you have or (2) tap into the crowd that already knows and understands `os.link`.

I'll leave it to you. :)

File: README.md (hunk 3 above)
Comment (mottosso) [reply README.md]: Or this?

```python
import filelink
filelink.create(src, dst, filelink.HARDLINK)
```

File: file_link/hardlink.py (hunk 2 above)
Comment (tokejepsen) [reply file_link/hardlink.py]: I'm going to leave it with "source" and "destination" :)

File: README.md (hunk 3 above)
Comment (tokejepsen) [reply README.md]: I like it :) Done!

File: filelink/link.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import platform
+
+
+HARDLINK = 0
+
+
+def _create_windows(source, destination, link_type):
+    """Creates hardlink at destination from source in Windows."""
+
+    if link_type == HARDLINK:
```
Comment (mottosso) [inline filelink/link.py:10]: Don't forget the `else` where the user passes e.g. `3` in which case it currently silently ignores it.

Comment (tokejepsen) [reply filelink/link.py:10]: Good point :) Done.

**Comment by mottosso** [comment]
Looks good! Giving this a try today.

File: README.md (hunk 3 above)
Comment (tokejepsen) [reply README.md]: Do we want to rename this repository to ```filelink```?

**Comment by mottosso** [comment]
Works as advertised! I think let's merge this and keep adding to it.

**Comment by tokejepsen** [comment]
Excellent!

**Pull request merged**